runtime/src/main/java/org/corfudb/recovery/FastSmrMapsLoader.java

```diff
@@ -183,12 +183,14 @@ public void updateStreamTails(long address, ILogData logData) {
         // On checkpoint, we also need to track the stream tail of the checkpoint
         if (isCheckPointEntry(logData)) {
             if (logData.getCheckpointType() == CheckpointEntry.CheckpointEntryType.END) {
-                streamTails.put(logData.getCheckpointedStreamId(),
-                        getStartAddressOfCheckPoint(logData));
+                streamTails.compute(logData.getCheckpointedStreamId(),
+                        (uuid, value) -> (value == null) ? getStartAddressOfCheckPoint(logData)
```

Comment:
@Maithem I don't understand your concern. are you saying that the log-address of the checkpoint should be considered? I don't see why, only the logical checkpoint start-address should matter.

In the example above, say that the checkpoint start-address is 2; hence 3 and 4 have -higher- log addresses than the checkpoint, although CP records are in higher log addresses (5, 6, 7).